### PR TITLE
Make signed http request exceptions serializable

### LINKED ISSUE #1324: Add the [Serializable] attribute to SignedHttpRequest exceptions
As of Wilson 6x we are planning on removing netstandard1.4 from the library.

By removing this target, we are now able to add the `[Serializable]` attribute back to many of our exceptions, as `[Serializable]` was not available for netstandard1.4.

We should add this attribute to `SignedHttpRequest` exceptions as well.

### PR DESCRIPTION
**Pull request by keegan-caruso**
Fixes #1324 

Add serializable attribute and deserialization ctor for signed http exceptions

### CODE REVIEW

**Review by GeoK — APPROVED** [review]
LGTM! Thanks Keegan.

Take a look at https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1300. It's similar to this item, but for the whole codebase.

File: test/Microsoft.IdentityModel.Protocols.SignedHttpRequest.Tests/SignedHttpRequestExceptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                var exception = Activator.CreateInstance(theoryData.ExceptionType);
+                var serializer = new DataContractSerializer(theoryData.ExceptionType);
+                var memoryStream = new MemoryStream();
+
+                // Validate that each exception is able to be both serialized and deserialized. If the exception is not able to be,
+                // an exception will be raised by the serializer object.
+                serializer.WriteObject(memoryStream, exception);
+                memoryStream.Seek(0, SeekOrigin.Begin);
+                _ = serializer.ReadObject(memoryStream);
+            }
```
Comment (GeoK) [inline test/Microsoft.IdentityModel.Protocols.SignedHttpRequest.Tests/SignedHttpRequestExceptionTests.cs:58]: Add `theoryData.ExpectedException.ProcessNoException(context);` here for consistency.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/How-We-Test#key-details

Comment (keegan-caruso) [reply test/Microsoft.IdentityModel.Protocols.SignedHttpRequest.Tests/SignedHttpRequestExceptionTests.cs:58]: Missed that. Thanks. Addressed in bbcc09f

**Pull request merged**